### PR TITLE
Add SNAP categorical eligibility and refactor other SNAP variables

### PR DESCRIPTION
**Pull request by MaxGhenis**
# New variables: several related to SNAP categorical eligibility

- [x] Label field added
- [x] Documentation field added
- [x] Unit field added
- [x] Default value field added if relevant
- [x] Variable name follows conventions
- [x] Unit test(s) added

## What's changed

Add variables and parameters for SNAP categorical eligibility, notably state-level BBCE which extends asset and income limits.

This also refactors the SNAP code, breaking variables into their own files and adding some intermediate variables with unit tests for clarity and debugging.

### CODE REVIEW

**Review by nikhilwoodruff — APPROVED** [review]
Looks good to merge to me, though maybe once merged we could file an issue for the commented out TANF categorical eligibility to keep track of it?

**Comment by MaxGhenis** [comment]
Filed #440 for TANF categorical eligibility and removed the comments. Will merge once the tests pass. Thanks!

**Pull request merged**
